pkgs/by-name/ue/uesave/package.nix

```diff
@@ -0,0 +1,32 @@
+{
+  fetchFromGitHub,
+  lib,
+  rustPlatform,
+  versionCheckHook,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "uesave";
+  version = "0.5.0";
+  src = fetchFromGitHub {
+    owner = "trumank";
+    repo = "uesave-rs";
+    rev = "v${version}";
+    hash = "sha256-9gOOSLejVfR1KJMhcNuKDkuTOvPC6sNG8xQOZlt8NxI=";
+  };
+
+  cargoHash = "sha256-U6RzSS2j6FK70OHlmWmHZZYT3UB0+Hi+uLofLy+XtGQ=";
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+  versionCheckProgram = "${placeholder "out"}/bin/${meta.mainProgram}";
```

Comment:
```suggestion
```

It's the same with `pname` so not needed

Reply:
What's the guideline on setting `meta.mainProgram`? Should we only set it when it's different from `pname`, or should we always set it?

Reply:
The current practice is to always set `meta.mainProgram`.

Reply:
Then, should we keep the practice consistent with `versionCheckProgram`?

Reply:
You are right, I would specifically discuss this with other maintainers.